code/game/machinery/Sleeper.dm

```diff
@@ -449,25 +453,23 @@
 /obj/machinery/sleeper/ex_act(severity)
 	if(filtering)
 		toggle_filter()
+	var/mob/M = src.occupant
 	switch(severity)
 		if(1.0)
-			for(var/atom/movable/A as mob|obj in src)
-				A.forceMove(src.loc)
-				ex_act(severity)
+			go_out()
+			M.ex_act(severity)
 			qdel(src)
 			return
 		if(2.0)
 			if(prob(50))
-				for(var/atom/movable/A as mob|obj in src)
-					A.forceMove(src.loc)
-					ex_act(severity)
+				go_out()
+				M.ex_act(severity)
 				qdel(src)
 				return
 		if(3.0)
 			if(prob(25))
-				for(var/atom/movable/A as mob|obj in src)
-					A.forceMove(src.loc)
-					ex_act(severity)

```

Comment:
That causes an infinite loop because it calls it's own ex_act(). The only reason the old version terminated at all was that that once everything was ejected, the for loop wasn't run at that level.